### PR TITLE
feat: view apps as any pub key

### LINKED ISSUE #2598: View as vega user
## Story

As a user
I want to view console/token as a specific pubkey
So that i can easily inspect chain data

## Acceptance Criteria

- [ ] I can enter a pubkey 
- [ ] I can specify a pubkey via a url param
- [ ] I can disconnect my pubkey

### PR DESCRIPTION
**Pull request by dexturr**
# Related issues 🔗

Closes #2598 

# Description ℹ️

Allows users to input a pubkey into token and console and then view the site as if they were that user.

### CODE REVIEW

File: apps/token/src/app-loader.tsx [hunk 1]
```diff
@@ -17,17 +17,28 @@ import {
 import { useContracts } from './contexts/contracts/contracts-context';
 import { useRefreshAssociatedBalances } from './hooks/use-refresh-associated-balances';
 import { Connectors } from './lib/vega-connectors';
+import { useSearchParams } from 'react-router-dom';
+
+const useUrlViewOnlyWalletConnect = () => {
```
Comment (dexturr) [inline apps/token/src/app-loader.tsx:22]: move into helper

File: apps/token/src/components/token-input/token-input.tsx [hunk 2]
```diff
@@ -100,6 +101,7 @@ export const TokenInput = ({
   approveText?: string;
   approveTxState?: TransactionState;
   approveTxDispatch?: React.Dispatch<TransactionAction>;
+  disabled?: boolean;
```
Comment (dexturr) [inline apps/token/src/components/token-input/token-input.tsx:104]: remove all disabled

File: apps/token/src/lib/vega-connectors.ts [hunk 3]
```diff
@@ -1,9 +1,17 @@
-import { RestConnector, JsonRpcConnector } from '@vegaprotocol/wallet';
+import {
+  RestConnector,
+  JsonRpcConnector,
+  ViewConnector,
+} from '@vegaprotocol/wallet';
+
+const urlParams = new URLSearchParams(window.location.search);
```
Comment (dexturr) [inline apps/token/src/lib/vega-connectors.ts:7]: do for console

File: apps/token/src/app-loader.tsx [hunk 4]
```diff
@@ -17,6 +17,18 @@ import {
 import { useContracts } from './contexts/contracts/contracts-context';
 import { useRefreshAssociatedBalances } from './hooks/use-refresh-associated-balances';
 import { Connectors } from './lib/vega-connectors';
+import { useSearchParams } from 'react-router-dom';
+
+const useVegaWalletEagerConnect = () => {
+  const vegaConnecting = useEagerConnect(Connectors);
+  const { pubKey, connect } = useVegaWallet();
+  const [searchParams] = useSearchParams();
+  const [query] = React.useState(searchParams.get('address'));
```
Comment (edd) [inline apps/token/src/app-loader.tsx:26]: **thought (blocking)**: I think we need to make this check that address is a valid public key. I chose to be difficult and use `network`, which is valid, but I think at least requiring it is x length and valid makes sense?

**Review by edd — APPROVED** [review]
I used this on console, and it worked nicely. However the 'Submit Order' button was still allowed (I was viewing as `network`. Is that a follow up ticket?

**Comment by dexturr** [comment]
> I used this on console, and it worked nicely. However the 'Submit Order' button was still allowed (I was viewing as `network`. Is that a follow up ticket?

@edd I discussed it with UX and their preference was that a message was shown on transaction attempt that says "You are connected in a view only state for public key: ${this.pubkey}. In order to send transactions you must connect to a real wallet."

So this is desired behaviour, moreover it makes it much easier to implement, if people want the buttons disabled we can do that as a follow up 👍

**Pull request merged**